packages/blocks/src/date-time-picker/index.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * External dependencies
+ */
+import { RawHTML, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import classnames from 'classnames';
+import { isEmpty } from 'lodash';
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+/**
+ * Internal dependencies
+ */
+import { useColorStyles } from '@crowdsignal/styles';
+import { ErrorMessage, FormInputWrapper } from '../components';
+import { useField } from '@crowdsignal/form';
+// import validator from './validations';
+
+const DateTimePicker = ( { attributes, className } ) => {
+	const [ startDate, setStartDate ] = useState( new Date() );
+	console.log( startDate );
+	const { error, onChange, fieldValue } = useField( {
+		fieldName: `q_${ attributes.clientId }[text]`,
+		validation: ( value ) => {
+			if ( attributes.mandatory && isEmpty( value ) ) {
+				return __( 'This field is required', 'blocks' );
+			}
+		},
+	} );
+
+	const classes = classnames(
+		className,
+		'crowdsignal-forms-date-time-picker-block',
+		{
+			'is-required': attributes.mandatory,
+			// 'is-error': error,
+		}
+	);
+
+	return (
+		<FormInputWrapper
+			className={ classes }
+			style={ { ...useColorStyles( attributes ) } }
+		>
+			<FormInputWrapper.Label className="crowdsignal-forms-text-input-block__label">
+				<RawHTML>{ attributes.label }</RawHTML>
+			</FormInputWrapper.Label>
+			<DatePicker
+				className={ classes }
+				selected={ startDate }
+				dateFormat="MMMM d, yyyy"
+				onChange={ ( date ) => setStartDate( date ) }
```

Comment:
Here you probably want to use the `fieldValue` property. 
This will link your block with the Redux store.

Then, to update the store value, you need to call the `onChange` handler exported by the `useField` hook.

```JavaScript
<DatePicker
	...
	selected={ fieldValue }
	onChange={ onChange }
	...
/>
```

If you need to initialize the block with some value, the `useField` hook accepts an `initialValue` property.
And, if you need to parse the selected date somehow, create a handler instead of passing the `onChange` directly to the `<DatePicket />` component.